internal/util/importutilv2/csv/row_parser.go

```diff
@@ -0,0 +1,438 @@
+		}
+		if len(vec) != r.dim {
+			return nil, r.wrapDimError(len(vec), field)
+		}
+		vec2 := make([]byte, len(vec)*2)
+		for i := 0; i < len(vec); i++ {
+			copy(vec2[i*2:], typeutil.Float32ToFloat16Bytes(float32(vec[i])))
+		}
+		return vec2, nil
+	case schemapb.DataType_BFloat16Vector:
+		var vec []float32
+		err := json.Unmarshal([]byte(obj), &vec)
+		if err != nil {
+			return nil, r.wrapTypeError(obj, field)
+		}
+		if len(vec) != r.dim {
+			return nil, r.wrapDimError(len(vec), field)
+		}
+		vec2 := make([]byte, len(vec)*2)
+		for i := 0; i < len(vec); i++ {
+			copy(vec2[i*2:], typeutil.Float32ToBFloat16Bytes(float32(vec[i])))
+		}
+		return vec2, nil
+	case schemapb.DataType_SparseFloatVector:
+		// use dec.UseNumber() to avoid float64 precision loss
+		var vec map[string]interface{}
+		dec := json.NewDecoder(strings.NewReader(obj))
+		dec.UseNumber()
+		err := dec.Decode(&vec)
+		if err != nil {
+			return nil, r.wrapTypeError(obj, field)
+		}
+		vec2, err := typeutil.CreateSparseFloatRowFromMap(vec)
+		if err != nil {
+			return nil, err
+		}
+		return vec2, nil
+	case schemapb.DataType_Array:
+		var vec []interface{}
+		desc := json.NewDecoder(strings.NewReader(obj))
+		desc.UseNumber()
+		err := desc.Decode(&vec)
+		if err != nil {
+			return nil, r.wrapTypeError(obj, field)
+		}
+		scalarFieldData, err := r.arrayToFieldData(vec, field.GetElementType())
+		if err != nil {
+			return nil, err
+		}
+		return scalarFieldData, nil
+	default:
+		return nil, merr.WrapErrImportFailed(fmt.Sprintf("parse csv failed, unsupport data type: %s",
+			field.GetDataType().String()))
+	}
+}
+
+func (r *rowParser) arrayToFieldData(arr []interface{}, eleType schemapb.DataType) (*schemapb.ScalarField, error) {
+	switch eleType {
+	case schemapb.DataType_Bool:
+		values := make([]bool, 0)
+		for i := 0; i < len(arr); i++ {
+			value, ok := arr[i].(bool)
+			if !ok {
+				return nil, r.wrapArrayValueTypeError(arr, eleType)
+			}
+			values = append(values, value)
+		}
+		return &schemapb.ScalarField{
+			Data: &schemapb.ScalarField_BoolData{
+				BoolData: &schemapb.BoolArray{
+					Data: values,
+				},
+			},
+		}, nil
+	case schemapb.DataType_Int8, schemapb.DataType_Int16, schemapb.DataType_Int32:
+		values := make([]int32, 0)
+		for i := 0; i < len(arr); i++ {
+			value, ok := arr[i].(json.Number)
+			if !ok {
+				return nil, r.wrapArrayValueTypeError(arr, eleType)
+			}
+			num, err := strconv.ParseInt(value.String(), 0, 32)
+			if err != nil {
+				return nil, err
+			}
+			values = append(values, int32(num))
+		}
+		return &schemapb.ScalarField{
+			Data: &schemapb.ScalarField_IntData{
+				IntData: &schemapb.IntArray{
+					Data: values,
+				},
+			},
+		}, nil
+	case schemapb.DataType_Int64:
+		values := make([]int64, 0)
+		for i := 0; i < len(arr); i++ {
+			value, ok := arr[i].(json.Number)
+			if !ok {
+				return nil, r.wrapArrayValueTypeError(arr, eleType)
+			}
+			num, err := strconv.ParseInt(value.String(), 0, 64)
```

Comment:
Let's make the `base` argument of `ParseInt` be 10 instead of 0, to avoid the leading 0 be falsely recognized as base of 8.